src/main/java/com/icthh/xm/tmf/ms/activation/service/SagaServiceImpl.java

```diff
@@ -233,6 +288,43 @@ public void retrySagaEvent(String txid, String eventId) {
         return transactionRepository.findAll(pageable);
     }
 
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void resendAllEventsInQueue() {
+        sagaEventRepository.findByStatus(IN_QUEUE).forEach(retryService::doResend);
+    }
+
+    @Override
+    public SagaTransaction findByKey(String key) {
```

Comment:
suggest to rename method to `getByKey` as it throws exception. 
in such a way we can create `findByKey` method returning optional in case of need.